parser-typechecker/src/Unison/Codebase/FileCodebase/Common.hs

```diff
@@ -313,30 +304,29 @@ branchFromFiles rootDir h = do
       Left  err   -> failWith $ InvalidEditsFile file err
       Right edits -> pure edits
 
--- returns Nothing if `root` has no root branch (in `branchHeadDir root`)
 getRootBranch :: forall m.
   MonadIO m => CodebasePath -> m (Either Codebase.GetRootBranchError (Branch m))
 getRootBranch root =
   ifM (codebaseExists root)
-    (listDirectory (branchHeadDir root) >>= go')
+    (listDirectory (branchHeadDir root) >>= filesToBranch)
     (pure $ Left Codebase.NoRootBranch)
  where
-  go' :: [String] -> m (Either Codebase.GetRootBranchError (Branch m))
-  go' = \case
-              []       -> pure $ Left Codebase.NoRootBranch
-              [single] -> runExceptT $ go single
-              conflict -> runExceptT (traverse go conflict) >>= \case
-                Right (x : xs) -> Right <$> foldM Branch.merge x xs
-                Right _ -> error "FileCodebase.getRootBranch.conflict can't be empty."
-                Left e -> Left <$> pure e
-
-  go :: String -> ExceptT Codebase.GetRootBranchError m (Branch m)
-  go single = case hashFromString single of
-    Nothing -> failWith $ CantParseBranchHead single
-    Just (Branch.Hash -> h)  -> ExceptT $ branchFromFiles root h <&> \case
-      Just b -> Right b
-      Nothing -> Left $ Codebase.CouldntLoadRootBranch h
-
+  filesToBranch :: [FilePath] -> m (Either Codebase.GetRootBranchError (Branch m))
+  filesToBranch = \case
+    []       -> pure $ Left Codebase.NoRootBranch
+    [single] -> runExceptT $ fileToBranch single
+    conflict -> runExceptT (traverse fileToBranch conflict) >>= \case
+      Right (x : xs) -> Right <$> foldM Branch.merge x xs
+      Right _ -> error "FileCodebase.getRootBranch.conflict can't be empty."
+      Left e -> Left <$> pure e
+
+  fileToBranch :: String -> ExceptT Codebase.GetRootBranchError m (Branch m)
+  fileToBranch single = ExceptT $ case hashFromString single of
+    Nothing -> pure . Left $ Codebase.CouldntParseRootBranch single
+    Just (Branch.Hash -> h) -> branchFromFiles root h <&>
+                                maybeToEither (Codebase.CouldntLoadRootBranch h)
```

Comment:
Just renamed these functions called `go`.  I think it's a slight improvement, but not huge.